self-run-agents/instances/azp-build-asg/iam.tf

```diff
@@ -5,6 +5,7 @@ data "aws_iam_policy_document" "asg_scale_in_protection" {
   statement {
     actions = [
       "autoscaling:SetInstanceProtection",
```

Comment:
Can't we remove this now that we're detaching them?

Reply:
Yes